### ISSUE TITLE
Discuss moving private keys inside client struct

### ISSUE THREAD

**Issue by vovapi** [asuleymanov/golos-go#14]
I want to discuss package-wide private key list:
https://github.com/asuleymanov/golos-go/blob/e0463ed5c878990a1fd1eb447881cdd4777c3819/client.go#L16

IMO this is bad design, because of:
* Package user cannot manipulate the process of signing, namely choose the key for signing, key is automatically.
* If you want to divide the concerns of your app, you could create different client instances for various tasks (e.g. posting operations & active operations), but with package-wide key-list this would not be possible
* In https://github.com/go-steem/rpc transactions are signed manually, so it is lacking this issue

In addition we could implement manual signing of transactions and export transaction broadcasting method for manually signed transactions.

**Comment by UncleAndy** [comment]
Согласен. @asuleymanov как смотришь на это? Тогда инициализацию ключей можно перенести в инициализацию клиента.

**Comment by asuleymanov** [comment]
правильно ли я Вас понял. Вы предлагаете еще сделать функции для операций с возможностью передачи ключа в них?

**Comment by UncleAndy** [comment]
Нет. Мы предлагаем убрать глобальную переменную KeyList и перенести ее в структуру Client.

**Comment by asuleymanov** [comment]
Я не сильно понимаю плюсы в данной реализации.
Но думаю что если два человека видят в этом плюс то стоит так и сделать.

**Comment by asuleymanov** [comment]
@vovapi.
Доработка произведенная @UncleAndy это то что нужно. или вы имели ввиду что то  другое?
